### PR TITLE
PHPCS: Infinite Scroll

### LINKED ISSUE #21965: PHPCS: Infinite Scroll
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
	"projects/plugins/jetpack/modules/infinite-scroll.php",
	"projects/plugins/jetpack/modules/infinite-scroll/infinity.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentyeleven.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentyfifteen.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentyfourteen.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentyseventeen.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentysixteen.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentyten.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentythirteen.php",
	"projects/plugins/jetpack/modules/infinite-scroll/themes/twentytwelve.php",
```

### PR DESCRIPTION
**Pull request by sdixon194**
Fixes #21965

#### Changes proposed in this Pull Request:

PHPCS Fixes for infinite scroll. fairly straightforwad changes, aside from some unsure nonce verification checks I ignored.

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

no
#### Testing instructions:

Proof, test infinite scroll using themes like twenty ten - twenty seventeen, try adding a footer and making sure it's disabled, etc.

### CODE REVIEW

File: projects/plugins/jetpack/modules/infinite-scroll/infinity.php [hunk 1]
```diff
@@ -1550,14 +1632,18 @@ function inject_query_args( $query_args ) {
...
+			$query_args,
+			array(
+				'suppress_filters' => false,
+			)
+		);
 
-		if ( is_array( $_REQUEST[ 'query_args' ] ) ) {
-			foreach ( $_REQUEST[ 'query_args' ] as $var => $value ) {
-				if ( in_array( $var, $allowed_vars ) && ! empty( $value ) )
+		if ( is_array( $_REQUEST['query_args'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotValidated -- no site changes.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/infinite-scroll/infinity.php:1642]:
```suggestion
		if ( isset( $_REQUEST['query_args'] ) && is_array( $_REQUEST['query_args'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- no site changes.
```

File: projects/plugins/jetpack/modules/infinite-scroll/infinity.php [hunk 2]
```diff
@@ -1550,14 +1632,18 @@ function inject_query_args( $query_args ) {
...
+			foreach ( $_REQUEST['query_args'] as $var => $value ) { // phpcs:ignore -- no site changes.
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/infinite-scroll/infinity.php:1643]:
```suggestion
			foreach ( wp_unslash( $_REQUEST['query_args'] ) as $var => $value ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- no site changes, sanitized below.
```

File: projects/plugins/jetpack/modules/infinite-scroll/infinity.php [hunk 3]
```diff
@@ -1299,32 +1373,37 @@ function ( $script_name ) use ( $initial_scripts ) {
...
+		// @todo see if we should validate this nonce since we use it to form a query.
+		$page = (int) $_REQUEST['page']; // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- we're casting this to an int and not making changes to the site.
 
 		// Sanitize and set $previousday. Expected format: dd.mm.yy
-		if ( preg_match( '/^\d{2}\.\d{2}\.\d{2}$/', $_REQUEST['currentday'] ) ) {
+		if ( preg_match( '/^\d{2}\.\d{2}\.\d{2}$/', $_REQUEST['currentday'] ) ) { // phpcs:ignore
 			global $previousday;
-			$previousday = $_REQUEST['currentday'];
+			$previousday = $_REQUEST['currentday']; // phpcs:ignore -- I don't think this is even used?
 		}
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/modules/infinite-scroll/infinity.php]: ba508d5636781c085dd6710b2206266f78cf243e

File: projects/plugins/jetpack/modules/infinite-scroll/infinity.php [hunk 4]
```diff
@@ -941,17 +987,21 @@ function action_wp_footer_settings() {
...
 		<script type="text/javascript">
 		//<![CDATA[
 		var infiniteScroll = JSON.parse( decodeURIComponent( '<?php echo
-			rawurlencode( json_encode( array( 'settings' => $js_settings ) ) );
+			rawurlencode( wp_json_encode( array( 'settings' => $js_settings ) ) );
 		?>' ) );
 		//]]>
 		</script>
 		<?php
+		// phpcs:enable -- end inline javascript.
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/modules/infinite-scroll/infinity.php]: ba508d5636781c085dd6710b2206266f78cf243e

File: projects/plugins/jetpack/modules/infinite-scroll/infinity.php [hunk 5]
```diff
@@ -425,14 +463,16 @@ function infinite_setting_html_calypso_placeholder() {
...
-	function infinite_setting_html() {
+	public function infinite_setting_html() {
 		$notice = '<em>' . __( 'We&rsquo;ve changed this option to a click-to-scroll version for you since you have footer widgets in Appearance &rarr; Widgets, or your theme uses click-to-scroll as the default behavior.', 'jetpack' ) . '</em>';
 
 		// If the blog has footer widgets, show a notice instead of the checkbox
-		if ( self::get_settings()->footer_widgets || 'click' == self::get_settings()->requested_type ) {
-			echo '<label>' . $notice . '</label>';
+		if ( self::get_settings()->footer_widgets || 'click' === self::get_settings()->requested_type ) {
+			// @todo find out if this is even rendered any more and escape it if so.
+			echo '<label>' . $notice . '</label>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- we know what the HTML is.
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/modules/infinite-scroll/infinity.php]: 741252910f9c540fd209f6d53bed44273bee4126

File: projects/plugins/jetpack/modules/infinite-scroll/infinity.php [hunk 6]
```diff
@@ -1181,11 +1243,12 @@ function ( $script_name ) use ( $initial_scripts ) {
...
 
-		if ( empty( $results['scripts'] ) )
-			unset( $results['scripts' ] );
+		if ( empty( $results['scripts'] ) ) {
+			unset( $results['scripts'] );
+		}
 
 		// Parse and sanitize the style handles already output
-		$initial_styles = isset( $_REQUEST['styles'] ) && is_array( $_REQUEST['styles'] ) ? array_map( 'sanitize_text_field', $_REQUEST['styles'] ) : false;
+		$initial_styles = isset( $_REQUEST['styles'] ) && is_array( $_REQUEST['styles'] ) ? array_map( 'sanitize_text_field', $_REQUEST['styles'] ) : false; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.MissingUnslash -- no site changes made, unslashing breaks styling.
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply projects/plugins/jetpack/modules/infinite-scroll/infinity.php]: Weird, I can't reproduce it now. tried a few different themes but seems fine. Previously it was screwing with the header and making it appear on the bottom of the page when scrolling, but maybe something else was weird that's now fixed ¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (anomiex) [reply projects/plugins/jetpack/modules/infinite-scroll/infinity.php]: So should we add the `wp_unslash()` then?

[the remark this answers is not in the document]
Comment (anomiex) [reply projects/plugins/jetpack/modules/infinite-scroll/infinity.php]: d'oh, GitHub's UI confused me. You did do that.

**Review by anomiex — APPROVED** [review]

**Comment by sdixon194** [comment]
r245728-wpcom

**Pull request merged**